pydoctor/templatewriter/writer.py

```diff
@@ -56,13 +57,28 @@ def prepOutputDirectory(self) -> None:
         """
         Write static CSS and JS files to build directory.
         """
-        os.makedirs(self.base, exist_ok=True)
-        for template in self.template_lookup.templates:
-            if isinstance(template, _StaticTemplate):
-                with self.base.joinpath(template.name).open('w', encoding='utf-8') as jobj:
-                    jobj.write(template.text)
+        os.makedirs(self.output_dir, exist_ok=True)
+        self._writeStaticTemplates(self.template_lookup.templates)
+    
+    def _writeStaticTemplates(self, templates: Iterable[Template], subfolder: Optional[str] = None) -> None:
+        """
+        Write all L{_StaticTemplate} to output directory, inspect L{_TemplateSubFolder} 
+        and reccursively write the static templates in subfolders.
+        """
+        _template_rel_path_t = f"{subfolder + os.sep if subfolder else ''}%s"
+        
+        for template in templates:
+            _template_rel_path = _template_rel_path_t % template.name
+            outfile = self.output_dir.joinpath(_template_rel_path)
+            if isinstance(template, _TemplateSubFolder):
+                os.makedirs(outfile, exist_ok=True)
+                self._writeStaticTemplates(template.lookup.templates, subfolder=_template_rel_path)
+                
+            elif isinstance(template, _StaticTemplate):
+                with outfile.open('w', encoding='utf-8') as fobj:
+                    fobj.write(template.text)
```

Comment:
Would it make sense to have these two fragments in a method on `Template`, implemented in both subclasses instead?

Reply:
You mean for instance to add a `write(path: Path)` method to `Template` superclass ? 

Then the subclass would implement it.